src/e3/testsuite/driver/diff.py

```diff
@@ -0,0 +1,314 @@
+import os
+import re
+import sys
+
+from e3.diff import diff
+from e3.os.fs import unixpath
+from e3.testsuite.result import Log, binary_repr
+from e3.testsuite.driver.classic import ClassicTestDriver, TestAbortWithError
+
+
+def indent(text, prefix="  "):
+    """prepend ``prefix`` to every line in ``text``.
+
+    :param str text: text to transform.
+    :param str prefix: string to prepend.
+    :rtype: str
+    """
+    return "\n".join((prefix + line) for line in text.split("\n"))
```

Comment:
splitlines

Reply:
This is not equivalent, for instance in the following case:
```
>>> "a\n".splitlines()
['a']
>>> "a\n".split("\n")
['a', '']
```

That trailing empty line matters here as it could be the reason the diff detects a mismatch, so we must preserve it. I’ll add a comment to make this explicit.

Reply:
Done